ui/app/models/ldap/config.js

```diff
@@ -0,0 +1,124 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Model, { attr } from '@ember-data/model';
+import { withFormFields } from 'vault/decorators/model-form-fields';
+import { withModelValidations } from 'vault/decorators/model-validations';
+
+const validations = {
+  binddn: [{ type: 'presence', message: 'Administrator Distinguished Name is required.' }],
+  bindpass: [{ type: 'presence', message: 'Administrator Password is required.' }],
+};
+const formGroups = [
+  { default: ['binddn', 'bindpass', 'url', 'password_policy'] },
+  { 'TLS options': ['starttls', 'insecure_tls', 'certificate', 'client_tls_cert', 'client_tls_key'] },
+  { 'More options': ['userdn', 'userattr', 'upndomain', 'connection_timeout', 'request_timeout'] },
+];
+
+@withModelValidations(validations)
+@withFormFields(null, formGroups)
+export default class KubernetesConfigModel extends Model {
+  @attr('string') backend; // dynamic path of secret -- set on response from value passed to queryRecord
+
+  @attr('string', {
+    label: 'Administrator Distinguished Name',
+    subText:
+      'Distinguished name of the adminstrator to bind (Bind DN) when performing user and group search. Example: cn=vault,ou=Users,dc=example,dc=com.',
+  })
+  binddn;
+
+  @attr('string', {
+    label: 'Administrator Password',
+    subText: 'Password to use along with Bind DN when performing user search.',
+  })
+  bindpass;
+
+  @attr('string', {
+    label: 'URL',
+    subText: 'The directory server to connect to.',
+  })
+  url;
+
+  @attr('string', {
+    editType: 'optionalText',
+    label: 'Use custom password policy',
+    subText: 'Specify the name of an existing password policy.',
+    defaultSubText: 'Unless a custom policy is specified, Vault will use a default.',
+    defaultShown: 'Default',
+    docLink: '/vault/docs/concepts/password-policies',
+  })
+  password_policy;
+
+  @attr('string') schema;
+
+  @attr('boolean', {
+    label: 'Start TLS',
+    subText: 'If checked, or address contains “ldaps://”, creates an encrypted connection with LDAP',
```

Comment:
To be consistent with the other subtext, can you end this with a period?

Reply:
Will do. I copied all of this from Figma so it must have been a bad highlight.